src/main/java/org/opensearch/knn/index/store/IndexOutputWithBuffer.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.store;
+
+import org.apache.lucene.store.IndexOutput;
+
+import java.io.IOException;
+
+public class IndexOutputWithBuffer {
+    // Underlying `IndexOutput` obtained from Lucene's Directory.
+    private IndexOutput indexOutput;
+    // Write buffer. Native engine will copy bytes into this buffer.
+    private byte[] buffer = new byte[64 * 1024];
```

Comment:
lets add a comment on why we are using this particular size as the buffer.

Reply:
sure, sounds good.